apps/desktop/src/autofill/services/desktop-fido2-user-interface.service.ts

```diff
@@ -50,17 +71,39 @@ export class DesktopFido2UserInterfaceSession implements Fido2UserInterfaceSessi
     private cipherService: CipherService,
     private accountService: AccountService,
     private logService: LogService,
+    private messagingService: MessagingService,
+    private router: Router,
+    private desktopSettingsService: DesktopSettingsService,
   ) {}
 
-  async pickCredential({
-    cipherIds,
-    userVerification,
-  }: PickCredentialParams): Promise<{ cipherId: string; userVerified: boolean }> {
-    this.logService.warning("pickCredential", cipherIds, userVerification);
+  pickCredential: (
+    params: PickCredentialParams,
+  ) => Promise<{ cipherId: string; userVerified: boolean }>;
+
+  private operationSubject = new Subject<void>();
+  private createdCipher: Cipher;
 
-    return { cipherId: cipherIds[0], userVerified: userVerification };
+  /**
+   * Notifies the Fido2UserInterfaceSession that the UI operations has completed and it can return to the OS.
+   */
+  notifyOperationCompleted() {
+    this.operationSubject.next();
+    this.operationSubject.complete();
   }
 
+  /**
+   * Returns once the UI has confirmed and completed the operation
+   * @returns
+   */
+  private async waitForUICompletion(): Promise<void> {
```

Comment:
nit: we don't treat acronyms/abbreviations differently (e.g. `Id` not `ID`)
```suggestion
  private async waitForUiCompletion(): Promise<void> {
```